File: leetcodejava/src/main/java/com/leetcode/SixSeven.java
package com.leetcode;

public class SixSeven {
    public String addBinary(String a, String b) {
        StringBuilder sb = new StringBuilder();
        int carry = 0;
        int i = a.length() - 1;
        int j = b.length() - 1;

        while (i >= 0 || j >= 0 || carry == 1) {
            if (i >= 0) {
                carry += a.charAt(i--) - '0';
            }
            if (j >= 0) {
                carry += b.charAt(j--) - '0';
            }

            sb.append(carry % 2);
            carry /= 2;
        }

        return sb.reverse().toString();
    }

    public static void main(String[] args) {
        SixSeven sixSeven = new SixSeven();
        System.out.println(sixSeven.addBinary("100", "110010"));
    }
}

File: leetcodecpp/leetcode/SixSeven.cpp
#include <string>
//
// Created by John on 2025/3/31.
//
using namespace std;

class Solution {
public:
    string addBinary(string a, string b) {
        // char[] aArr = a.toCharArray(), bArr = b.toCharArray();
        string res;
        int i = a.length() - 1;
        int j = b.length() - 1;
        int carry = 0;
        while (i >= 0 || j >= 0 || carry >= 0) {
            int sum = carry;
            if (j >= 0) {
                carry += b[j--] - '0';
            }
            if (i >= 0) {
                carry += a[i--] - '0';
            }
            res += to_string(sum % 2);
            sum /= 2;
        }
        reverse(res.begin(), res.end());
        return res;
    }
};